apps/server/src/modules/common-cartridge/service/common-cartridge-export.service.ts

```diff
@@ -1,43 +1,273 @@
+import { RichTextElementResponseDto } from '../common-cartridge-client/card-client/dto/rich-text-element-response.dto';
+import { LinkElementResponseDto } from '../common-cartridge-client/card-client/dto/link-element-response.dto';
+
+const isRichTextElement = (reference: unknown): reference is RichTextElementResponseDto =>
+	reference instanceof RichTextElementResponseDto;
+
+const isLinkElement = (reference: unknown): reference is LinkElementResponseDto =>
+	reference instanceof LinkElementResponseDto;
 
 @Injectable()
 export class CommonCartridgeExportService {
 	constructor(
-		private readonly filesService: FilesStorageClientAdapterService,
 		private readonly boardClientAdapter: BoardClientAdapter,
 		private readonly cardClientAdapter: CardClientAdapter,
 		private readonly coursesClientAdapter: CoursesClientAdapter,
-		private readonly courseRoomsClientAdapter: CourseRoomsClientAdapter
+		private readonly courseRoomsClientAdapter: CourseRoomsClientAdapter,
+		private readonly lessonClinetAdapter: LessonClientAdapter,
+		private readonly mapper: CommonCartridgeExportMapper
 	) {}
 
-	public async findCourseFileRecords(courseId: string): Promise<FileDto[]> {
-		const courseFiles = await this.filesService.listFilesOfParent(courseId);
+	private async findCourseCommonCartridgeMetadata(courseId: string): Promise<CourseCommonCartridgeMetadataDto> {
+		const courseCommonCartridgeMetadata = await this.coursesClientAdapter.getCourseCommonCartridgeMetadata(courseId);
 
-		return courseFiles;
+		return courseCommonCartridgeMetadata;
 	}
 
-	public async findCourseCommonCartridgeMetadata(courseId: string): Promise<CourseCommonCartridgeMetadataDto> {
-		const courseCommonCartridgeMetadata = await this.coursesClientAdapter.getCourseCommonCartridgeMetadata(courseId);
+	private async findRoomBoardByCourseId(courseId: string): Promise<RoomBoardDto> {
+		const roomBoardDto = await this.courseRoomsClientAdapter.getRoomBoardByCourseId(courseId);
 
-		return courseCommonCartridgeMetadata;
+		return roomBoardDto;
 	}
 
-	public async findRoomBoardByCourseId(courseId: string): Promise<RoomBoardDto> {
-		const courseRooms = await this.courseRoomsClientAdapter.getRoomBoardByCourseId(courseId);
+	private async findBoardSkeletonById(boardId: string): Promise<BoardSkeletonDto> {
+		const boardSkeleton = await this.boardClientAdapter.getBoardSkeletonById(boardId);
 
-		return courseRooms;
+		return boardSkeleton;
 	}
 
-	public async findAllCardsByIds(ids: Array<string>): Promise<CardListResponseDto> {
+	private async findAllCardsByIds(ids: Array<string>): Promise<CardListResponseDto> {
 		const cards = await this.cardClientAdapter.getAllBoardCardsByIds(ids);
 
 		return cards;
 	}
+
+	private async findLessonById(lessonId: string): Promise<LessonDto> {
+		const lesson = await this.lessonClinetAdapter.getLessonById(lessonId);
+
+		return lesson;
+	}

```

Comment:
Move the private methods below the public methods. Are these methods necessary?

Reply:
done ✅